bug: Transformer crashes on markdown tables

Currently, the transformer crashes when trying to parse a markdown table with the following error message:

```console
./tools/transformer/transformer docs/examples/sandbox-transformer-walk-through.md sandbox-developer/sandbox-transformer-walk-through
panic: runtime error: index out of range [28] with length 20

goroutine 1 [running]:
github.com/yuin/goldmark/renderer.(*renderer).Render.func2({0x1052d2848, 0x140004a1cc0}, 0x1)
        /Users/jayclifford/go/pkg/mod/github.com/yuin/goldmark@v1.7.2/renderer/renderer.go:164 +0xc8
github.com/yuin/goldmark/ast.walkHelper({0x1052d2848, 0x140004a1cc0}, 0x1400063fa80)
        /Users/jayclifford/go/pkg/mod/github.com/yuin/goldmark@v1.7.2/ast/ast.go:492 +0x34
github.com/yuin/goldmark/ast.walkHelper({0x1052d0e68, 0x140004b4cf0}, 0x1400063fa80)
        /Users/jayclifford/go/pkg/mod/github.com/yuin/goldmark@v1.7.2/ast/ast.go:498 +0x8c
github.com/yuin/goldmark/ast.Walk(...)
        /Users/jayclifford/go/pkg/mod/github.com/yuin/goldmark@v1.7.2/ast/ast.go:487
github.com/yuin/goldmark/renderer.(*renderer).Render(0x14000182eb0, {0x1052ce348, 0x14000056028}, {0x140003b6000, 0x2e4e, 0x3000}, {0x1052d0e68, 0x140004b4cf0})
        /Users/jayclifford/go/pkg/mod/github.com/yuin/goldmark@v1.7.2/renderer/renderer.go:161 +0x1c8
main.writeFile({0x1052cf340, 0x140003bf840}, {0x16ae2b6f2, 0x32}, {0x14000375430, 0x8}, {0x140003b6000, 0x2e4e, 0x3000})
        /Users/jayclifford/Repos/killercoda/tools/transformer/main.go:74 +0x1d4
main.transform({0x16ae2b6c0, 0x31}, {0x16ae2b6f2, 0x32})
        /Users/jayclifford/Repos/killercoda/tools/transformer/main.go:172 +0xaa8
main.main()
        /Users/jayclifford/Repos/killercoda/tools/transformer/main.go:52 +0x174
```

This was discovered when trying to transform: [Tranformer Walkthrough](https://github.com/grafana/killercoda/blob/staging/docs/examples/sandbox-transformer-walk-through.md#metadata) into a tutorial. 

Here is the table in question:
```markdown
| Field                                    | Type   | Description                                                                                                                                                            |
| ---------------------------------------- | ------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `killercoda.backend.imageid`             | String | The name of the Killercoda environment's backend image. Supported values include `ubuntu`.                                                                             |
| `killercoda.description`                 | String | The description displayed on the Killercoda website                                                                                                                    |
| `killercoda.details.finish.text`         | String | The filename of the finish page Markdown source in the `grafana/killercoda` repository. A [finish directive](#finish) in the documentation source overrides this.      |
| `killercoda.details.intro.text`          | String | The filename of the introduction page Markdown source in the `grafana/killercoda` repository. An [intro directive](#intro) in the documentation source overrides this. |
| `killercoda.preprocessing.substitutions` | Array  | Substitute matches of a regular expression with a replacement. For more information, refer to [Substitutions](#substitutions).                                         |
| `killercoda.title`                       | String | The title for the tutorial on the Killercoda website.                                                                                                                  |
```

--- a/tools/transformer/goldmark/extension/table.go
+++ b/tools/transformer/goldmark/extension/table.go
@@ -1,0 +1,101 @@
+// Package extension implements Goldmark extensions.
+package extension
+
+import (
+	"github.com/yuin/goldmark"
+	gast "github.com/yuin/goldmark/ast"
+	"github.com/yuin/goldmark/extension"
+	"github.com/yuin/goldmark/extension/ast"
+	"github.com/yuin/goldmark/renderer"
+	"github.com/yuin/goldmark/util"
+)
+
+// TableMarkdownRenderer is a renderer.NodeRenderer implementation that
+// renders Table nodes.
+type TableMarkdownRenderer struct {
+	extension.TableConfig
+}
+
+// NewTableMarkdownRenderer returns a new TableMarkdownRenderer.
+func NewTableMarkdownRenderer(opts ...extension.TableOption) renderer.NodeRenderer {
+	r := &TableMarkdownRenderer{
+		TableConfig: extension.NewTableConfig(),
+	}
+	for _, opt := range opts {
+		opt.SetTableOption(&r.TableConfig)
+	}
+	return r
+}
+
+// RegisterFuncs implements renderer.NodeRenderer.RegisterFuncs.
+func (r *TableMarkdownRenderer) RegisterFuncs(reg renderer.NodeRendererFuncRegisterer) {
+	reg.Register(ast.KindTable, r.renderTable)
+	reg.Register(ast.KindTableHeader, r.renderTableHeader)
+	reg.Register(ast.KindTableRow, r.renderTableRow)
+	reg.Register(ast.KindTableCell, r.renderTableCell)
+}
+
+func (r *TableMarkdownRenderer) renderTable(w util.BufWriter, source []byte, n gast.Node, entering bool) (gast.WalkStatus, error) {
+	if !entering {
+		_, _ = w.WriteString("\n")
+	}
+
+	return gast.WalkContinue, nil
+}
+
+func (r *TableMarkdownRenderer) renderTableHeader(w util.BufWriter, source []byte, n gast.Node, entering bool) (gast.WalkStatus, error) {
+	if !entering {
+		_, _ = w.WriteString("\n")
+
+		// Add a separator line after the header row.
+		_, _ = w.WriteString("|")
+		for i := 0; i < n.ChildCount(); i++ {
+			_, _ = w.WriteString(" --- |")
+		}
+		_, _ = w.WriteString("\n")
+	}
+
+	return gast.WalkContinue, nil
+}
+
+func (r *TableMarkdownRenderer) renderTableRow(w util.BufWriter, _ []byte, n gast.Node, entering bool) (gast.WalkStatus, error) {
+	if !entering {
+		_, _ = w.WriteString("\n")
+	}
+
+	return gast.WalkContinue, nil
+}
+
+func (r *TableMarkdownRenderer) renderTableCell(w util.BufWriter, _ []byte, n gast.Node, entering bool) (gast.WalkStatus, error) {
+	if entering {
+		_, _ = w.WriteString("| ")
+	} else if n.NextSibling() != nil {
+		_, _ = w.WriteString(" ")
+	} else {
+		_, _ = w.WriteString(" |")
+	}
+
+	return gast.WalkContinue, nil
+}
+
+type table struct {
+	options []extension.TableOption
+}
+
+// Table is an extension that allow you to use GFM tables .
+var Table = &table{
+	options: []extension.TableOption{},
+}
+
+// NewTable returns a new extension with given options.
+func NewTable(opts ...extension.TableOption) goldmark.Extender {
+	return &table{
+		options: opts,
+	}
+}
+
+func (e *table) Extend(m goldmark.Markdown) {
+	m.Renderer().AddOptions(renderer.WithNodeRenderers(
+		util.Prioritized(NewTableMarkdownRenderer(e.options...), 500),
+	))
+}

--- a/tools/transformer/main.go
+++ b/tools/transformer/main.go
@@ -12,6 +12,7 @@ import (
 	"path/filepath"
 	"regexp"
 
+	"github.com/grafana/killercoda/tools/transformer/goldmark/extension"
 	"github.com/grafana/killercoda/tools/transformer/killercoda"
 	"github.com/yuin/goldmark"
 	"github.com/yuin/goldmark/text"
@@ -91,7 +92,9 @@ func transform(srcFilePath, dstDirPath string) error {
 	md := goldmark.New(goldmark.WithExtensions(&KillercodaExtension{
 		Transformers:        DefaultKillercodaTransformers,
 		AdditionalExtenders: []goldmark.Extender{},
-	}))
+	},
+		extension.Table,
+	))
 
 	root := md.Parser().Parse(text.NewReader(data))
 
@@ -127,7 +130,9 @@ func transform(srcFilePath, dstDirPath string) error {
 			AdditionalExtenders: []goldmark.Extender{
 				&StepTransformer{StartMarker: pageIntroStartMarker, EndMarker: pageIntroEndMarker},
 			},
-		}))
+		},
+			extension.Table,
+		))
 
 		if err := writeFile(md, dstDirPath, "intro.md", data); err != nil {
 			return err
@@ -142,7 +147,9 @@ func transform(srcFilePath, dstDirPath string) error {
 			AdditionalExtenders: []goldmark.Extender{
 				&StepTransformer{StartMarker: pageFinishStartMarker, EndMarker: pageFinishEndMarker},
 			},
-		}))
+		},
+			extension.Table,
+		))
 
 		if err := writeFile(md, dstDirPath, "finish.md", data); err != nil {
 			return err
@@ -167,7 +174,9 @@ func transform(srcFilePath, dstDirPath string) error {
 				AdditionalExtenders: []goldmark.Extender{
 					&StepTransformer{StartMarker: startMarker, EndMarker: endMarker},
 				},
-			}))
+			},
+				extension.Table,
+			))
 
 			if err := writeFile(md, dstDirPath, fmt.Sprintf("step%d.md", i), data); err != nil {
 				errs = errors.Join(errs, err)

